main.cpp

```diff
@@ -64,151 +63,592 @@ New/This/Pointers/References conclusion
 {
-    return lhs + rhs;
+    std::cout << "(dt) " << *value << " + (dt) " << *dt.value << " = ";
+    *value = *value + *dt.value;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-int IntType::subtract(int lhs, int rhs)
+DoubleType& DoubleType::subtract(const DoubleType& dt)
 {
-    return lhs - rhs;
+    std::cout << "(dt) " << *value << " - (dt) " << *dt.value << " = ";
+    *value = *value - *dt.value;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-int IntType::multiply(int lhs, int rhs)
+DoubleType& DoubleType::multiply(const DoubleType& dt)
 {
-    return lhs * rhs;
+    std::cout << "(dt) " << *value << " * (dt) " << *dt.value << " = ";
+    *value = *value * *dt.value;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-int IntType::divide(int lhs, int rhs)
+DoubleType& DoubleType::divide(const DoubleType& dt)
 {
-    if (rhs == 0)
+    if (*dt.value == 0)
     {
-        std::cout << "Error: cannot divide by 0! Returning 0 instead\n";
-        return 0;
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(dt) " << *value << " / (dt) " << *dt.value << " = ";
+    *value = *value / *dt.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// DoubleType member functions implementation
+// IntType argument
+DoubleType& DoubleType::add(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " + (it) " << *it.value << " = ";
+    *value = *value + *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::subtract(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " - (it) " << *it.value << " = ";
+    *value = *value - *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::multiply(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " * (it) " << *it.value << " = ";
+    *value = *value * *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::divide(const IntType& it)
+{
+    if (*it.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(dt) " << *value << " / (it) " << *it.value << " = ";
+    *value = *value / *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// Primitives
+IntType& IntType::add(int v)
+{
+    std::cout << "(it) " << *value << " + " << v << " = ";
+    *value += v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(int v)
+{
+    std::cout << "(it) " << *value << " - " << v << " = ";
+    *value -= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(int v)
+{
+    std::cout << "(it) " << *value << " * " << v << " = ";
+    *value *= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(int v)
+{
+    if (v == 0.0f)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
```

Comment:
`v` is an int, yet you're comparing it with a `float`

You also didn't prevent the division from happening.